### ISSUE TITLE
Restore compatibility with Java 8

### ISSUE THREAD

**Issue by f-cramer** [ezylang/EvalEx#451]
I would really like to use version 3, but unfortunatelly I am stuck with Java 8 for the foreseeable future. May I ask what was the reason for requiring Java 11 starting with version 3?

I already made the necessary changes to restore compatibility with Java 8 in [my fork](https://github.com/f-cramer/EvalEx). It did not seem that any "new" API were used other then the `of` constructors on `List` and `Set` and `Map#entries`. Those can easily be changed to `Collections#singleton(List)`, `Arrays#asList` and `new AbstractMap.SimpleEntry`. Other then that I only had to downgrade `googleJavaFormat` to version `1.7`, because newer versions require Java 11.

**Comment by uklimaschewski** [comment]
Hi, I took the chance of the new major version with all the breaking changes to also upgrade the Java version.
Though I was also thinking about moving to Java 17, I decided to be a bit more conservative and had chosen Java 11.
In general, I would like to continue to also upgrade to newer Java versions from time to time.

When it comes to Java 8, there are possibly also some hidden problems, apart from the language changes.
In version 2 (which uses Java 8), I had to add special treatments for the Java module system, which was introduced in Java 9.
